qualtran/bloqs/factoring/ecc/ec_add.py

```diff
@@ -30,6 +50,939 @@
 )
 from qualtran.bloqs.mod_arithmetic._shims import ModInv
 from qualtran.resource_counting import BloqCountDictT, SympySymbolAllocator
+from qualtran.simulation.classical_sim import ClassicalValT
+from qualtran.symbolics.types import HasLength
+
+from .ec_point import ECPoint
+
+
+@frozen
+class _ECAddStepOne(Bloq):
+    r"""Performs step one of the ECAdd bloq.
+
+    Args:
+        n: The bitsize of the two registers storing the elliptic curve point
+        mod: The modulus of the field in which we do the addition.
+        uncompute: whether to compute or uncompute.
+
+    Registers:
+        f1: Flag to set if a = x.
+        f2: Flag to set if b = -y.
+        f3: Flag to set if (a, b) = (0, 0).
+        f4: Flag to set if (x, y) = (0, 0).
+        ctrl: Flag to set if neither the input points nor the output point are (0, 0).
+        a: The x component of the first input elliptic curve point of bitsize `n`.
+        b: The y component of the first input elliptic curve point of bitsize `n`.
+        x: The x component of the second input elliptic curve point of bitsize `n`, which
+           will contain the x component of the resultant curve point.
+        y: The y component of the second input elliptic curve point of bitsize `n`, which
+           will contain the y component of the resultant curve point.
+
+    References:
+        [How to compute a 256-bit elliptic curve private key with only 50 million Toffoli gates](https://arxiv.org/abs/2306.08585)
+        Fig 10
+    """
+
+    n: int
+    mod: int
+    uncompute: bool = False
+
+    @cached_property
+    def signature(self) -> 'Signature':
+        side = Side.LEFT if self.uncompute else Side.RIGHT
+        return Signature(
+            [
+                Register('f1', QBit(), side=side),
+                Register('f2', QBit(), side=side),
+                Register('f3', QBit(), side=side),
+                Register('f4', QBit(), side=side),
+                Register('ctrl', QBit(), side=side),
+                Register('a', QMontgomeryUInt(self.n)),
+                Register('b', QMontgomeryUInt(self.n)),
+                Register('x', QMontgomeryUInt(self.n)),
+                Register('y', QMontgomeryUInt(self.n)),
+            ]
+        )
+
+    def adjoint(self) -> '_ECAddStepOne':
+        return attrs.evolve(self, uncompute=self.uncompute ^ True)
+
+    def on_classical_vals(
+        self, a: 'ClassicalValT', b: 'ClassicalValT', x: 'ClassicalValT', y: 'ClassicalValT'
```

Comment:
the classical action of the adjoint will fail since the adjoint has those sided register which the on_classica_vals doesn't expect

Reply:
I removed adjoint stuff - it's not required at all for this circuit and I think it'd be better to focus my efforts on other things for now.